### PR TITLE
optimize: make host match ~15% faster

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
nice PR ^_^

**Comment by nic-chen** [comment]
good idea

**Review by nic-chen — APPROVED** [review]

File: lib/resty/radixtree.lua [hunk 1]
```diff
@@ -258,9 +273,7 @@ function pre_insert_route(self, path, route)
             local is_wildcard = false
             if h and h:sub(1, 1) == '*' then
                 is_wildcard = true
-                h = h:sub(2):reverse()
-            else
-                h = h:reverse()
+                h = h:sub(2)
```
[the remark this answers is not in the document]
Comment (spacewander) [reply lib/resty/radixtree.lua:276]: @nic-chen 
Remove the '*' prefix.

[the remark this answers is not in the document]
Comment (nic-chen) [reply lib/resty/radixtree.lua:276]: my mistake, the `else` in the old code affects my view.

**Comment by nic-chen** [comment]
merged, thanks a lot. @spacewander

**Pull request merged**